### ISSUE TITLE
Pull private containers

### ISSUE THREAD

**Issue by noose** [docker/swarm#2270]
Hi!
I want to pull/push some private containers (hosted on hub.docker.com). But I can't.

I have cluster with 2 nodes and 2 managers.
```bash
% export DOCKER_HOST='....:4000';
% docker login
Login with your Docker ID to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com to create one.
Username: noose
Password:
Login Succeeded

% docker pull noose/my-container:latest
ip-10-0-1-206: Pulling noose/my-container:latest... : Error: image noose/my-container not found
ip-10-0-1-207: Pulling noose/my-container:latest... : Error: image noose/my-container not found
Error: image noose/my-container:latest not found
```

How it's possible that swarm is production-ready, but there are such errors?

**Comment by riuvshin** [comment]
It is known issue https://github.com/docker/swarm/issues/2217

**Comment by nishanttotla** [comment]
Closing this issue since discussion is tracked in #2217 and fixed in #2272
